randutil/random.go

```diff
@@ -22,80 +19,89 @@ func init() {
+	if _, err := rand.Read(bytes); err != nil {
+		panic("rand.Reader returned an error")
 	}
-	return bytes, nil
+	return bytes
 }
 
 // String returns a random string of a given length using the characters in
 // the given string. It splits the string on runes to support UTF-8
 // characters.
-func String(length int, chars string) (string, error) {
+func String(length int, chars string) string {
 	result := make([]rune, length)
 	runes := []rune(chars)
 	x := int64(len(runes))
 	for i := range result {
+		// Since Go 1.24, it is guaranteed to never return an error, see https://github.com/golang/go/issues/66821.
+		// This error check can be removed once we drop support for Go 1.23.
 		num, err := rand.Int(rand.Reader, big.NewInt(x))
 		if err != nil {
-			return "", errors.Wrap(err, "error creating random number")
+			panic("rand.Reader returned an error")
 		}
 		result[i] = runes[num.Int64()]
 	}
-	return string(result), nil
+	return string(result)
 }
 
 // Hex returns a random string of the given length using the hexadecimal
 // characters in lower case (0-9+a-f).
-func Hex(length int) (string, error) {
+func Hex(length int) string {
 	return String(length, "0123456789abcdef")
 }
 
 // Alphanumeric returns a random string of the given length using the 62
 // alphanumeric characters in the POSIX/C locale (a-z+A-Z+0-9).
-func Alphanumeric(length int) (string, error) {
+func Alphanumeric(length int) string {
 	return String(length, "abcdefghijklmnopqrstuvwxyz0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ")
 }
 
 // ASCII returns a securely generated random ASCII string. It reads random
 // numbers from crypto/rand and searches for printable characters. It will
 // return an error if the system's secure random number generator fails to
 // function correctly, in which case the caller must not continue.
-func ASCII(length int) (string, error) {
+func ASCII(length int) string {
 	return String(length, ascii)
 }
 
 // Alphabet returns a random string of the given length using the 52
 // alphabetic characters in the POSIX/C locale (a-z+A-Z).
-func Alphabet(length int) (string, error) {
+func Alphabet(length int) string {
 	return String(length, "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ")
 }
 
 // UUIDv4 returns the string representation of a UUID version 4. Because 6 bits
 // are used to indicate the version 4 and the variant 10, the randomly generated
 // part has 122 bits.
-func UUIDv4() (string, error) {
+func UUIDv4() string {
 	var uuid [16]byte
-	_, err := io.ReadFull(rand.Reader, uuid[:])
-	if err != nil {
-		return "", errors.Wrap(err, "error generating uuid")
+	// crypto/rand.Reader is guaranteed to never return an error, and always fill the entire slices
+	// crypto/rand.Reader always fill the entire slices.
+	// Since Go 1.24, it is guaranteed to never return an error, see https://github.com/golang/go/issues/66821.
+	// This error check can be removed once we drop support for Go 1.23.
```

Comment:
```suggestion
	// crypto/rand.Reader is guaranteed to never return an error, and always fill the entire slices
	// crypto/rand.Reader always fills the entire slice.
	// Since Go 1.24, it is guaranteed to never return an error, see https://github.com/golang/go/issues/66821.
	// This error check can be removed once we drop support for Go 1.23.
```